### DISCUSSION TITLE
How to identify the event source accurately?

### DISCUSSION THREAD

**Discussion by dulajra** [h2oai/wave#820] [Q&A] [answered]
I have 3 dropdowns on the same screen. 2nd dropdown is populated based on the selection of 1st dropdown and the 3rd dropdown is populated based on the selection of the 2nd dropdown. The selection of each dropdown is saved in `q.client` state to preserve the selection during the trigger. The problem arises with this use case.

### Problem

The user has selected values for all 3 dropdowns. Then the user goes and changes the value of the 1st dropdown. In this case, `q.args` will have values set for all 3 dropdowns. Hence we cannot identify the source of the event (which dropdown was changed with the latest event). 

Here is the minimal code for my 3 dropdowns.

```
items.append(
        ui.dropdown(
            name='project',
            label='Select a Project',
            value=q.args.project,
            choices=get_project_choices(),
            trigger=True
        ),
    )

    if q.args.project:
        items.append(
            ui.dropdown(
                name='model',
                label='Select a Model',
                value=q.args.model,
                choices=get_model_choices(q.args.project),
                trigger=True
            )
        )

    if q.args.project and q.args.model:
        items.append(
            ui.dropdown(
                name='deployment',
                label='Select a Deployment',
                value=q.args.deployment,
                choices=_get_deployment_choices(q.args.project, q.args.model),
                trigger=True
            )
        )
```

### Expected behaviour
I want to reset and hide the 2nd and 3rd dropdowns if the user changes the value of the 1st dropdown again. But I cannot do this because I don't which dropdown was changed because all 3 have values set in `q.args`

**Comment by vopani** [comment, marked as answer]
@dulajra I have the same use-case where I check `change in value` instead.

Something like:

```python
if q.args.dropdown1 and q.args.dropdown1 != q.client.dropdown1:
    q.client.dropdown1 = q.args.dropdown1
    reset 2nd and 3rd dropdown
elif q.args.dropdown2 and q.args.dropdown2 != q.client.dropdown2:
    q.client.dropdown2 = q.args.dropdown2
    reset 3rd dropdown
```

**Reply by dulajra** [reply]
Great suggestion! Thanks @vopani
